connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SamplingTestPlugin.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.runtime.isolation;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Mix-in interface for plugins so we can sample information about their initialization
+ */
+public interface SamplingTestPlugin {
+
+    /**
+     * @return the ClassLoader used to statically initialize this plugin class
+     */
+    ClassLoader staticClassloader();
+
+    /**
+     * @return the ClassLoader used to initialize this plugin instance
+     */
+    ClassLoader classloader();
+
+    /**
+     * @return a group of other SamplingTestPlugin instances known by this plugin
+     * This should only return direct children, and not reference this instance directly
+     */
+    default Map<String, SamplingTestPlugin> otherSamples() {
```

Comment:
Default methods are not going to port backward past `2.0`, since default methods are a Java 8 feature. To work with Java 7, you'd have to make this an abstract class.

I'm not sure it's worth that change, since there are quite a few other things that will likely have to change when backporting. For example, `HeaderConverter` was introduced in AK 1.1, and `ConfigProvider` was introduced in AK 2.0. Removing these won't be trivial, so that means we're probably going to need separate PRs for some of these older branches.

Reply:
I was able to change this to an abstract class fairly easily. I hope that this makes backporting easier, and all you have to do for the incompatibilities is delete part of this patch.